src/ms3/logger.py

```diff
@@ -15,24 +20,27 @@
     'C': logging.CRITICAL,
 }
 
-DEFAULT_LOG_FORMAT = '%(levelname)-8s %(name)s -- %(message)s'
 
-def get_default_formatter():
-    format = DEFAULT_LOG_FORMAT
-    return logging.Formatter(format)
+class MessageType(Enum):
+    """Enumerated constants of message types."""
+    NO_TYPE = 0  # 0 is reserved as no type message
+    MCS_NOT_EXCLUDED_FROM_BARCOUNT_WARNING = 1
+    INCORRECT_VOLTA_MN_WARNING = 2
+    INCOMPLETE_MC_WRONGLY_COMPLETED_WARNING = 3
+    VOLTAS_WITH_DIFFERING_LENGTHS_WARNING = 4
+    MISSING_END_REPEAT_WARNING = 5
+    SUPERFLUOUS_TONE_REPLACEMENT_WARNING = 6
 
-class ContextAdapter(logging.LoggerAdapter):
-    """ This LoggerAdapter is designed to include the module and function that called the logger."""
-    def process(self, msg, overwrite={}, stack_info=False, **kwargs):
-        # my_context = kwargs.pop('my_context', self.extra['my_context'])
-        fn, l, f, s = self.logger.findCaller(stack_info=stack_info)
-        fname = os.path.basename(overwrite.pop('fname', fn))
-        line = overwrite.pop('line', l)
-        func = overwrite.pop('func', f)
-        stack = overwrite.pop('stack', s)
-        msg = msg.replace('\n', '\n\t')
-        message = f"{fname} (line {line}) {func}():\n\t{msg}" if stack is None else f"{fname} line {line}, {func}():\n\t{msg}\n{stack}"
-        return message, kwargs
+
+class CustomFormatter(logging.Formatter):
+    """Formats message depending on whether there is a specified message type"""
+    def format(self, record):
+        if record._message_type == 0:  # if there is no message type
```

Comment:
Can we assume that the message type is always 0 if it hasn't been specified? Maybe it would be safer that we first test for non-existence.